gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/HadoopFileSystemTestBase.java

```diff
@@ -401,6 +401,29 @@ public void testWrite1Byte() throws IOException {
     assertWithMessage("testWrite1Byte: write-read mismatch").that(readText).isEqualTo(text);
   }
 
+  /** Validates HTTP Get Statistics */
+  public void testHttpStatistics() throws IOException {
+    URI path = GoogleCloudStorageFileSystemIntegrationTest.getTempFilePath();
+    Path hadoopPath = ghfsHelper.castAsHadoopPath(path);
+    String text = "Hello World!";
+    int numBytesWritten = ghfsHelper.writeFile(hadoopPath, text, 1, /* overwrite= */ false);
+
+    try (FSDataInputStream readStream = ghfs.open(hadoopPath)) {
+
+      //  Check the statistics of method read().
+      readStream.read();
+      readStream.close();
+      assertThat(readStream.getIOStatistics().counters().get(STREAM_READ_OPERATIONS)).isEqualTo(1);
+    }
+    ghfs.close();
+    assertThat(
+            ((GoogleHadoopFileSystem) ghfs)
+                .getIOStatistics()
+                .counters()
+                .get(GhfsStatistic.ACTION_HTTP_GET_REQUEST.getSymbol()))
+        .isEqualTo(4);
```

Comment:
Are there 4 GET requests in this case ? Could you add a comment on which requests are sent ?